File: tests/contracts.py
"""Loads the canonical queue-tab column contract from tests/contracts.json.

Same source the node harness reads (docs/CONTRACTS.md), so the python worker's
reader column layout is checked against the exact same contract the backend writer
is checked against. Cross-check test = TEST item #4.
"""
import json
import os

_HERE = os.path.dirname(os.path.abspath(__file__))

with open(os.path.join(_HERE, "contracts.json"), encoding="utf-8") as _f:
    _RAW = json.load(_f)

PROVISIONING_QUEUE_COLUMNS = _RAW["provisioning_queue_columns"]
OFFBOARDING_QUEUE_COLUMNS = _RAW["offboarding_queue_columns"]
PROVISIONING_STATUS = _RAW["provisioning_status"]
PROVISIONING_SYSTEMS = _RAW["provisioning_systems"]
PROVISIONING_ACTIONS = _RAW["provisioning_actions"]
OFFBOARDING_STATUS = _RAW["offboarding_status"]
INLINE_SYSTEMS = _RAW["inline_systems"]
WORKER_SYSTEMS = _RAW["worker_systems"]
SYSTEMS_MASTER = _RAW["systems_master"]
OFFBOARDING_SYSTEMS_DEFAULT = _RAW["offboarding_systems_default"]
ENTITIES = _RAW["entities"]
MAX_ATTEMPTS_DEFAULT = _RAW["max_attempts_default"]
FIRE_DELAY_MS = _RAW["fire_delay_ms"]
QUEUE_ID_FORMAT = _RAW["queue_id_format"]
OFFB_ID_FORMAT = _RAW["offb_id_format"]

File: tests/contracts.js
// Loads the canonical queue-tab column contract from tests/contracts.json, which
// is transcribed from docs/CONTRACTS.md (the authoritative frozen wire). Both this
// node module and tests/contracts.py read the SAME json, so the cross-check test
// (TEST item #4) compares the real backend writer and python worker reader
// against one shared source. If CONTRACTS.md changes, update contracts.json only.
//
// Column order matters: index 0 == column A, index 1 == column B, etc.

const fs = require('fs');
const path = require('path');

const raw = JSON.parse(
  fs.readFileSync(path.join(__dirname, 'contracts.json'), 'utf8')
);

module.exports = {
  PROVISIONING_QUEUE_COLUMNS: raw.provisioning_queue_columns,
  OFFBOARDING_QUEUE_COLUMNS: raw.offboarding_queue_columns,
  PROVISIONING_STATUS: raw.provisioning_status,
  PROVISIONING_SYSTEMS: raw.provisioning_systems,
  PROVISIONING_ACTIONS: raw.provisioning_actions,
  OFFBOARDING_STATUS: raw.offboarding_status,
  INLINE_SYSTEMS: raw.inline_systems,
  WORKER_SYSTEMS: raw.worker_systems,
  SYSTEMS_MASTER: raw.systems_master,
  OFFBOARDING_SYSTEMS_DEFAULT: raw.offboarding_systems_default,
  ENTITIES: raw.entities,
  MAX_ATTEMPTS_DEFAULT: raw.max_attempts_default,
  FIRE_DELAY_MS: raw.fire_delay_ms,
  QUEUE_ID_FORMAT: raw.queue_id_format,
  OFFB_ID_FORMAT: raw.offb_id_format,
};
